pkg/controller/baremetalhost/baremetalhost_controller_test.go

```diff
@@ -192,6 +195,82 @@ func waitForProvisioningState(t *testing.T, r *ReconcileBareMetalHost, host *met
 	)
 }
 
+// TestStatusAnnotation_EmptyStatus ensures that status is manually populated
+// when status annotation is present and status field is empty.
+func TestStatusAnnotation_EmptyStatus(t *testing.T) {
+	host := newDefaultHost(t)
+	host.Annotations = map[string]string{
+		metal3v1alpha1.StatusAnnotation: statusAnnotaion,
+	}
+	host.Spec.Online = true
+	host.Spec.Image = &metal3v1alpha1.Image{URL: "foo", Checksum: "123"}
+
+	r := newTestReconciler(host)
+
+	tryReconcile(t, r, host,
+		func(host *metal3v1alpha1.BareMetalHost, result reconcile.Result) bool {
+			if host.Status.HardwareProfile == "StatusProfile" && host.Status.Provisioning.Image.URL == "bar" {
+				return true
+			}
+			return false
+		},
+	)
+}
+
+// TestStatusAnnotation_StatusPresent tests that if status is present
+// status annotation is ignored.
+func TestStatusAnnotation_StatusPresent(t *testing.T) {
+	host := newDefaultHost(t)
+	host.Annotations = map[string]string{
+		metal3v1alpha1.StatusAnnotation: statusAnnotaion,
+	}
+	host.Spec.Online = true
+	time := metav1.Now()
+	host.Status.LastUpdated = &time
+	host.Status.Provisioning.Image = metal3v1alpha1.Image{URL: "foo", Checksum: "123"}
+	r := newTestReconciler(host)
+
+	tryReconcile(t, r, host,
+		func(host *metal3v1alpha1.BareMetalHost, result reconcile.Result) bool {
+			if host.Status.HardwareProfile != "StatusProfile" && host.Status.Provisioning.Image.URL == "foo" {
+				return true
+			}
+			return false
+		},
+	)
+}
+
+// TestStatusAnnotation tests if statusAnnotaion is populated correctly
+func TestStatusAnnotation(t *testing.T) {
+	host := newDefaultHost(t)
+	host.Spec.Online = true
+	host.Spec.Image = &metal3v1alpha1.Image{URL: "foo", Checksum: "123"}
+	bmcSecret := newBMCCredsSecret(defaultSecretName, "User", "Pass")
+	r := newTestReconciler(host, bmcSecret)
+
+	tryReconcile(t, r, host,
+		func(host *metal3v1alpha1.BareMetalHost, result reconcile.Result) bool {
+			if utils.StringInList(host.Finalizers, metal3v1alpha1.BareMetalHostFinalizer) {
+				return true
+			}
+			return false
+		},
+	)
```

Comment:
why are you testing the finalizers ? Is it just so that you go through the loop twice because the first time it only sets the finalizer ? If so, shouldn't this be the case also for the other function, i.e. are you sure that the previous function is actually testing the function you intend to test ? I have a suspicion that it is not the case as you have two different ways to do things in two functions that should be essentially the same.
Also, you are not actually checking that the status was not modified. what this does is testing that the annotation matches the status, independently of the starting point of the annotation.